src/io/inputs/yml-system/converter.cpp

```diff
@@ -101,7 +101,7 @@ static Component createComponent(const YmlSystem::Component& c,
 {
     const auto [libraryId, modelId] = splitLibraryModelString(c.model);
 
-    const Model& model = getModel(libraries, libraryId, modelId);
+    Model& model = getModel(libraries, libraryId, modelId);
```

Comment:
Note a non-const ref can be converted to `const`. A `const` ref can't be converted to non-const, except using `const_cast`.

You probably don't need to remove the `const` qualifier here.

Reply:
There would be a solution to all of this **const** problematic.

Reply:
Have you examined the case of injecting optimConfig before libraries are created?

Reply:
Yes, but this way does not allow checking that all variables / objectives in the **optim-config.yml** exist in one of the libraries yml files